server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPRoutesHandler.java

```diff
@@ -62,13 +64,12 @@ boolean matches(HttpServerRequest request) {
     }
 
     Version extractRequestVersionHeader(HttpServerRequest request) {
-        return Optional.ofNullable(request.requestHeaders().get(ACCEPT))
-            .map(s -> s.split(";"))
-            .map(Arrays::stream)
-            .orElse(Stream.of())
-            .map(value -> value.trim().toLowerCase())
-            .filter(value -> value.startsWith(JMAP_VERSION_HEADER.toLowerCase()))
-            .map(value -> value.substring(JMAP_VERSION_HEADER.length()))
+        return request.requestHeaders()
+            .getAll(ACCEPT)
+            .stream()
+            .flatMap(element -> Arrays.stream(BasicHeaderValueParser.parseParameters(element, null)))
+            .filter(nameValuePair -> nameValuePair.getName().toLowerCase().startsWith(JMAP_VERSION_HEADER.toLowerCase()))
```

Comment:
why do you call `toLowerCase` on a constant defined locally? it looks like a waste

Reply:
I removed that in the fixup after